lib/runner/create-item-context.js

```diff
@@ -0,0 +1,50 @@
+var _ = require('lodash'),
+    sdk = require('postman-collection'),
+
+    SAFE_CONTEXT_PROPERTIES = ['replayState', 'coords'],
+
+    /**
+     * Creates a context object to be used with `http-request.command` extension.
+     *
+     * @param {Object} payload
+     * @param {Item} payload.item
+     * @param {Object} [payload.coords]
+     * @param {Object} [defaults]
+     * @param {Object} [defaults.replayState]
+     * @param {Object} [defaults.coords]
+     *
+     * @returns {Object}
+     */
+    createItemContext = function (payload, defaults) {
+        // extract properties from defaults that can/should be reused in new context
+        var context = defaults ? _.pick(defaults, SAFE_CONTEXT_PROPERTIES) : {};
+
+        // set cursor to context
+        !context.coords && (context.coords = payload.coords);
+
+        // save original item for reference
+        context.originalItem = payload.item;
+
+        // we clone item from the payload, so that we can make any changes we need there, without mutating the
+        // collection
+        context.item = new sdk.Item(payload.item.toJSON());
```

Comment:
Thinking out loud: There should be a clone method in the SDK 🤔

Reply:
Maybe 🤔